frontend/src/components/ide/ProgramIde.tsx

```diff
@@ -90,25 +84,23 @@ const ProgramIde: React.FC<{
   assertIsDefined(userProgram);
 
   // These values come from wasm. They're read only, so they're safe to share
-  // with the whole component tree. They are pointed and there updates won't
-  // trigger re-renders, but these values shouldn't be changing while this
-  // component tree is mounted.
-  const wasmHardwareSpec = useMemo(
+  // with the whole component tree. They are pointers and therefore updates
+  // won't trigger re-renders, but these values shouldn't be changing while
+  // this component tree is mounted anyway.
+  const wasmHardwareSpec = useStaticValue(
     () =>
       new HardwareSpec(
         hardwareSpec.numRegisters,
         hardwareSpec.numStacks,
         hardwareSpec.maxStackLength
-      ),
-    [hardwareSpec]
+      )
   );
-  const wasmProgramSpec = useMemo(
+  const wasmProgramSpec = useStaticValue(
     () =>
       new ProgramSpec(
         Int16Array.from(programSpec.input),
         Int16Array.from(programSpec.expectedOutput)
-      ),
-    [programSpec]
+      )
   );
```

Comment:
I also realized I could use our new `useStaticValue` hook here too. The React team have said that in the future, they may change `useMemo` to call the memoized function more than necessary if it's more performant, which in our case might cause some fuqqery. Have it be guaranteed stable should be a bit safer.